docs/snippets/pre_commit.rst

```diff
@@ -0,0 +1,31 @@
+**pydocstyle** can be included as a hook for `pre-commit`_.  The easiest way to get
+started is to add this configuration to your ``.pre-commit-config.yaml``:
+
+.. code-block:: yaml
+
+    -   repo: https://github.com/pycqa/pydocstyle
+        rev: ''  # pick a git hash / tag to point to
+        hooks:
+        -   id: pydocstyle
+
+See the `pre-commit docs`_ for how to customize this configuration.
+
+Checked-in python files will be passed as positional arguments so no need to use ``--match=*.py``.
+You can also use command line arguments instead of configuration files
+to achieve the same effect with less files.
+
+.. code-block:: yaml
+
+    - id: pydocstyle
+      args:
+      - --ignore=D100,D203,D405
+      # or multiline
+      - |-
+              --select=
+              D101,
+              D2
```

Comment:
I'd drop this second half (from line 12 onwards) -- the configuration file is a better place to configure this than with `args:`

Reply:
I prefer less configuration files though if I could have all in one.

Reply:
but yeah no prob ;) the first part is straight forward